### ISSUE TITLE
Meta-ticket for S07 and update of List implementation details

### ISSUE THREAD

**Issue by pmichaud** [perl6/specs#70]
tl;dr:  This is a meta-ticket collecting references to outstanding issues related to Perl 6's aggregate list types.

Full version:  Next week at the 2014 Austrian Perl Workshop I'm planning to focus heavily on updating Synopsis 7 and resolving a large number of issues surrounding Lists in Perl 6.

Having done at least three or four separate List implementations in Rakudo, I can tell you that it has never been easy to isolate -- small changes to the design of List (and its related classes) often have huge impacts to large segments of the codebase.  Lists seem to be incredibly fundamental to Perl 6 design, and it's vital to get the underlying structures "just right" .  (By way of analogy: Perl 6 OO design required getting the underlying metamodel details "just right", as it affects the entire implementation.)

To prepare for the update, I'd like to collect as many unresolved issues and details regarding S07 and lists into this single "metaticket".  These can be references to existing bug reports, IRC discussions, and the like.  Code examples illustrating undesirable behaviors can also be added here.  

Note: If a code example illustrates a bug or issue, find/file a separate ticket for it and then reference it here, please.  Also, this ticket is simply for collecting references to issues -- discussion of issues should take place elsewhere (either in separate tickets or on #perl6, which can then be linked to here).

Topics that I know are intimately involved in Perl 6 list design include (watch this section for updates):
- list-y types:  List, Array, Parcel, LoL, Iterable, Iterator, Range, Nil
- eager vs lazy
- finite vs infinite
- flattening vs non-flattening
- hyper, race
- core functions/methods:  map  sort  elems  eager  zip  tree
- core operators:  .[]  ...  Z\*  ,  «_»  [_]
- statements:  for  gather  take

So, any unresolved items relating to any of the above are fair game here.

Thanks.

**Comment by FROGGS** [comment]
Would be interesting to know how this microbenchmark will improve due to GLR:
https://p6weekly.wordpress.com/2015/02/17/2015-07-i-like-trains-fast-trains-that-is/comment-page-1/#comment-293
```
perl -wnl -E ‘say $1 if /\b(\w{5})\b/’ logs.txt (19Mb) / 1 sec
ruby -wnl -e ‘puts $1 if /\b(\w{5})\b/’ logs.txt (19Mb) / 1.8 secs
perl6 -n -e ‘say $0 if m/(<<\w**5>>)/’ logs.txt (19Mb) / 88 secs
```

**Comment by skids** [comment]
FWIW, the return value of a while/until/loop statement is a longstanding NYI.  And feeds have lost most of whatever laziness they may have had a while back.
